### DISCUSSION TITLE
Cant upload image to storage through code it always gice network erro

### DISCUSSION THREAD

**Discussion by iori2kimoz** [supabase/supabase#20321] [Questions]
Hi all , i am trying to take a photo with an app and then upload it to the storage it always give me network error or some code i tried uplaod it with zero bytes . can any one let me know where i need to fix this or where am going wrong ?

here is the code 

```typescript
import React, { useState, useEffect } from "react";
import {
  StyleSheet,
  View,
  Button,
  Image,
  TextInput,
  Alert,
} from "react-native";
import * as ImagePicker from "expo-image-picker";
import * as Location from "expo-location";
import { supabase } from "../../config/initSupabase";
import * as FileSystem from "expo-file-system";
// import { ImageManipulator } from "expo-image-manipulator";
import * as ImageManipulator from "expo-image-manipulator";

const CheckInScreen: React.FC = () => {
  const [image, setImage] = useState<string | null>(null);
  const [location, setLocation] = useState<any>(null);
  const [justification, setJustification] = useState<string>("");
  const [showJustification, setShowJustification] = useState<boolean>(false);

  useEffect(() => {
    (async () => {
      const { status: cameraStatus } =
        await ImagePicker.requestCameraPermissionsAsync();
      if (cameraStatus !== "granted") {
        Alert.alert("Permission needed", "Camera permission is required");
      }
    })();
  }, []);

  // const takePhoto = async () => {
  //   let result = await ImagePicker.launchCameraAsync({
  //     mediaTypes: ImagePicker.MediaTypeOptions.Images,
  //     // allowsEditing: true,
  //     // aspect: [4, 3],
  //     // quality: 1,
  //   });

  //   if (!result.canceled && result.assets && result.assets.length > 0) {
  //     setImage(result.assets[0].uri);
  //     await fetchLocation();
  //   }
  // };

  const takePhoto = async () => {
    let result = await ImagePicker.launchCameraAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.Images,
      // other options...
    });

    if (!result.canceled && result.assets && result.assets.length > 0) {
      const { uri } = result.assets[0];

      // Resize the image
      const resizedImage = await ImageManipulator.manipulateAsync(
        uri,
        [{ resize: { width: 500 } }], // Set the width you desire
        { compress: 0.5, format: ImageManipulator.SaveFormat.JPEG } // Adjust compression and format as needed
      );

      setImage(resizedImage.uri);
      await fetchLocation();
    }
  };

  const fetchLocation = async () => {
    let { status } = await Location.requestForegroundPermissionsAsync();
    if (status !== "granted") {
      Alert.alert("Permission needed", "Location permission is required");
      return;
    }

    let location = await Location.getCurrentPositionAsync({});
    setLocation(location);
  };

  const submitCheckIn = async () => {
    const timestamp = new Date().toISOString();
    const data = {
      uri: image,
      justification,
      location: `${location.coords.latitude}, ${location.coords.longitude}`,
      time: timestamp,
    };

    if (new Date().getHours() >= SPECIFIC_HOUR) {
      setShowJustification(true);
      return;
    }

    const { data: response, error } = await supabase
      .from("checkin")
      .insert([data]);
    if (error) {
      console.error("Error inserting data:", error);
      return;
    }

    if (image) {
      try {
        const fileUri = image;
        const fileInfo = await FileSystem.getInfoAsync(fileUri);

        if (!fileInfo.exists) {
          console.error("File does not exist");
          return;
        }

        const response = await fetch(fileUri);
        const blobData = await response.blob();

        // Upload the Blob to Supabase storage
        await supabase.storage
          .from("files")
          .upload(`photo_${timestamp}.png`, blobData);

        console.log("Image upload successful");
      } catch (error) {
        console.error("Error uploading image:", error);
      }
    }

    // if (image) {
    //   try {
    //     const fileUri = image;
    //     const fileInfo = await FileSystem.getInfoAsync(fileUri);

    //     if (!fileInfo.exists) {
    //       console.error("File does not exist");
    //       return;
    //     }

    //     const fileData = await FileSystem.readAsStringAsync(fileUri, {
    //       encoding: FileSystem.EncodingType.Base64,
    //     });

    //     // Decode Base64 to binary data using Buffer in React Native
    //     const binaryData = Buffer.from(fileData, "base64");

    //     // Create a Blob from binary data
    //     const blobData = new Blob([binaryData], { type: "image/jpeg" });

    //     // Upload the Blob to Supabase storage
    //     await supabase.storage
    //       .from("files")
    //       .upload(`photo_${timestamp}.jpg`, blobData);

    //     console.log("Image upload successful");
    //   } catch (error) {
    //     console.error("Error uploading image:", error);
    //   }
    // }
    //upload but file is 0 bytes
    // if (image) {
    //   try {
    //     const response = await supabase.storage
    //       .from("files")
    //       .upload(`photo_${timestamp}.jpg`, { file: image });
    //     console.log("Image upload response:", response);
    //   } catch (error) {
    //     console.error("Error uploading image:", error);
    //   }
    // }

    //not uploading from mobile but uploading from web :
    // if (image) {
    //   const file = new Blob([await (await fetch(image)).blob()], {
    //     type: "image/jpeg",
    //   });
    //   await supabase.storage
    //     .from("files")
    //     .upload(`photo_${timestamp}.jpg`, file);
    //   try {
    //     const response = await supabase.storage
    //       .from("files")
    //       .upload(`photo_${timestamp}.jpg`, file);
    //     console.log("Image upload response:", response);
    //   } catch (error) {
    //     console.error("Error uploading image:", error);
    //   }
    // }

    setImage(null);
    setLocation(null);
    setJustification("");
    setShowJustification(false);
  };

  return (
    <View style={styles.container}>
      <Button title="Take Photo" onPress={takePhoto} />
      {image && <Image source={{ uri: image }} style={styles.image} />}
      <Button title="Submit Check-In" onPress={submitCheckIn} />
      {showJustification && (
        <TextInput
          placeholder="Enter justification"
          value={justification}
          onChangeText={(text) => setJustification(text)}
        />
      )}
    </View>
  );
};

const SPECIFIC_HOUR = 22; // Change this to your specific hour

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    padding: 20,
  },
  image: {
    width: 300,
    height: 300,
    resizeMode: "contain",
    marginVertical: 20,
  },
});

export default CheckInScreen;

``` 

here is the error : 
 ERROR  Error uploading image: [TypeError: Network request failed]

and here is the log on supabase :
 Log ID
55287356-9295-4368-be0e-386806f91352

Log Timestamp (UTC)
2024-01-10T19:43:13.538Z

Log Event Message
POST | 400 | 156.208.39.45 | 843768593dea73a3 | https://ejuwoeewarptvyxhliwi.supabase.co/storage/v1/object/files/photo_2024-01-10T19:43:11.861Z.png | okhttp/4.9.2

Log Metadata
[
  {
    "logflare_worker": [
      {
        "worker_id": "HNY18C"
      }
    ],
    "request": [
      {
        "cf": [
          {
            "asOrganization": "Telecom Egypt",
            "asn": 8452,
            "botManagement": [
              {
                "corporateProxy": false,
                "detectionIds": [],
                "ja3Hash": null,
                "jsDetection": [
                  {
                    "passed": false
                  }
                ],
                "score": 99,
                "staticResource": false,
                "verifiedBot": false
              }
            ],
            "city": "Giza",
            "clientAcceptEncoding": "gzip",
            "clientTcpRtt": 70,
            "clientTrustScore": null,
            "colo": "MRS",
            "continent": "AF",
            "country": "EG",
            "edgeRequestKeepAliveStatus": 1,
            "httpProtocol": "HTTP/2",
            "isEUCountry": null,
            "latitude": "29.99790",
            "longitude": "31.04830",
            "metroCode": null,
            "postalCode": null,
            "region": "Giza",
            "regionCode": "GZ",
            "requestPriority": "weight=16;exclusive=0;group=0;group-weight=0",
            "timezone": "Africa/Cairo",
            "tlsCipher": "AEAD-AES128-GCM-SHA256",
            "tlsClientAuth": [
              {
                "certPresented": "0",
                "certRevoked": "0",
                "certVerified": "NONE"
              }
            ],
            "tlsExportedAuthenticator": [
              {
                "clientFinished": "c74b143aaa4be37b66db9da7770b885215d5980b3dab7f2c9855051be244d46d",
                "clientHandshake": "c25e91a066df5b8d636d7d3912802e0a5b91dffa0f5a6b41293ff3d74aad1cdf",
                "serverFinished": "53ca350d4c25699d05c7058c81cd2a4e7a5cf312600da671bf677f0652309e0f",
                "serverHandshake": "de519c2ae9f781ee9507fc7557342dd04581907a3fdce7b8a6536f50ad3aefb7"
              }
            ],
            "tlsVersion": "TLSv1.3",
            "verifiedBotCategory": null
          }
        ],
        "headers": [
          {
            "accept": null,
            "cf_cache_status": null,
            "cf_connecting_ip": "156.208.39.45",
            "cf_ipcountry": "EG",
            "cf_ray": "843768593dea73a3",
            "content_length": "162",
            "content_location": null,
            "content_range": null,
            "content_type": "multipart/form-data; boundary=bdc20055-ef8f-4dd5-8531-e93074ac9fa0",
            "date": null,
            "host": "ejuwoeewarptvyxhliwi.supabase.co",
            "prefer": null,
            "range": null,
            "referer": null,
            "sb_gateway_mode": null,
            "sb_gateway_version": null,
            "user_agent": "okhttp/4.9.2",
            "x_client_info": "supabase-js-react-native/2.39.2",
            "x_forwarded_proto": "https",
            "x_forwarded_user_agent": null,
            "x_kong_proxy_latency": null,
            "x_kong_upstream_latency": null,
            "x_real_ip": "156.208.39.45"
          }
        ],
        "host": "ejuwoeewarptvyxhliwi.supabase.co",
        "method": "POST",
        "path": "/storage/v1/object/files/photo_2024-01-10T19:43:11.861Z.png",
        "port": null,
        "protocol": "https:",
        "sb": [
          {
            "auth_user": "21b3a256-0e52-4743-990c-0e0a49aa32f9"
          }
        ],
        "search": null,
        "url": "https://ejuwoeewarptvyxhliwi.supabase.co/storage/v1/object/files/photo_2024-01-10T19:43:11.861Z.png"
      }
    ],
    "response": [
      {
        "headers": [
          {
            "cf_cache_status": "DYNAMIC",
            "cf_ray": "8437685977c873a3-MRS",
            "content_length": "84",
            "content_location": null,
            "content_range": null,
            "content_type": "application/json; charset=utf-8",
            "date": "Wed, 10 Jan 2024 19:43:13 GMT",
            "sb_gateway_mode": "direct",
            "sb_gateway_version": "1",
            "transfer_encoding": null,
            "x_kong_proxy_latency": null,
            "x_kong_upstream_latency": null
          }
        ],
        "origin_time": 697,
        "status_code": 400
      }
    ]
  }
]

**Comment by Pedro141006** [comment]
Hi, you resolve this problem, i have the same problem.

**Reply by iori2kimoz** [reply]
I solved it yes i followed this tutorials and i had an issue with the AUTH provider if i remember correctly 

https://www.youtube.com/watch?v=am6w5zEDk_g&list=PPSV

**Reply by Pedro141006** [reply]
Thanks, i will try this.
